### ISSUE TITLE
Refactor  search / search stream

### ISSUE THREAD

**Issue by fulmicoton** [quickwit-oss/quickwit#1750]
We have several issues.

Right now the search stream logic is needlessly convoluted.
Also see #1744 #1745 #1749.

A lot of the logic is redundant.

One final key problem is that the logic we use to detect whether fast field are useful or not happens without fast field meta information.

**Comment by fmassot** [comment]
I think this can be closed thanks to #1752 or did you have more in mind @fulmicoton ?

**Comment by fulmicoton** [comment]
Yes I think we can have a deeper look at this. The current code is still very redundant / not very good...

(I added #1693 to the ticket to the list of ticket to consider )
